python/lsst/ap/pipe/createApFakes.py

```diff
@@ -0,0 +1,300 @@
+    """Config for CreateRandomApFakesTask. Copy from the InsertFakesConfig to
+    assert that columns created with in this task match that those expected in
+    the InsertFakes and related tasks.
+    """
+    fakeDensity = pexConfig.RangeField(
+        doc="Goal density of random fake sources per square degree. Default "
+            "value is roughly the density per square degree for ~10k sources "
+            "visit.",
+        dtype=float,
+        default=1000,
+        min=0,
+    )
+    filterSet = pexConfig.ListField(
+        doc="Set of Abstract filter names to produce magnitude columns for.",
+        dtype=str,
+        default=["u", "g", "r", "i", "z", "y"],
+    )
+    fraction = pexConfig.RangeField(
+        doc="Fraction of the created source that should be inserted into both "
+            "the visit and template images. Values less than 1 will result in "
+            "(1 - fraction) / 2 inserted into only visit or the template.",
+        dtype=float,
+        default=1/3,
+        min=0,
+        max=1,
+    )
+    magMin = pexConfig.RangeField(
+        doc="Minimum magnitude the mag distribution. All magnitudes requested "
+            "are set to the same value.",
+        dtype=float,
+        default=20,
+        min=1,
+        max=40,
+    )
+    magMax = pexConfig.RangeField(
+        doc="Maximum magnitude the mag distribution. All magnitudes requested "
+            "are set to the same value.",
+        dtype=float,
+        default=30,
+        min=1,
+        max=40,
+    )
+    randomSeed = pexConfig.Field(
+        doc="Random seed to set for reproducible datasets",
+        dtype=int,
+        default=None,
+    )
+    visitSourceFlagCol = pexConfig.Field(
+        doc="Name of the column flagging objects for insertion into the visit "
+            "image.",
+        dtype=str,
+        default="isVisitSource"
+    )
+    templateSourceFlagCol = pexConfig.Field(
+        doc="Name of the column flagging objects for insertion into the "
+            "template image.",
+        dtype=str,
+        default="isTemplateSource"
+    )
+
+
+class CreateRandomApFakesTask(PipelineTask):
+    """Create and store a set of spatially uniform fakes for use in AP
+    processing.
+    """
+
+    _DefaultName = "createApFakes"
+    ConfigClass = CreateRandomApFakesConfig
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+        inputs["tractId"] = butlerQC.quantum.dataId["tract"]
+
+        outputs = self.run(**inputs)
+        butlerQC.put(outputs, outputRefs)
+
+    def run(self, tractId, skyMap):
+        """Create a set of uniform random points that covers a tract.
+
+        Parameters
+        ----------
+        tractId : `int`
+            Tract id to produce randoms over.
+        skyMap : `lsst.skymap.SkyMap`
+            Skymap to produce randoms over.
+
+        Returns
+        -------
+        randoms : `lsst.pipe.tasks.PaquertTable`
```

Comment:
`PaquertTable` -> `ParquetTable`

Reply:
It's actually returning DataFrame now so I will change it.